### PR TITLE
Fix #5094 : [Wiki update] Bazel build failed in mac m1

### LINKED ISSUE #5094: [BUG]: Bazel build failed in mac m1
### Describe the bug

This bug is faced by me when I updated my Mac OS from Monterey to Ventura, I got this error while building Bazel:

```
ERROR: /Users/OpenSource/oppia-android/model/src/main/proto/BUILD.bazel:167:20: Generating JavaLite proto_library //model/src/main/proto:profile_proto failed: (Segmentation fault): protoc failed: error executing command bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/protoc '--proto_path=bazel-out/android-armeabi-v7a-fastbuild/bin/model/src/main/proto/_virtual_imports/languages_proto' ... (remaining 8 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox protoc failed: error executing command bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/protoc '--proto_path=bazel-out/android-armeabi-v7a-fastbuild/bin/model/src/main/proto/_virtual_imports/languages_proto' ... (remaining 8 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
Target //:oppia failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 70.242s, Critical Path: 12.83s
INFO: 282 processes: 21 internal, 261 darwin-sandbox.
FAILED: Build did NOT complete successfully
```

### Describe the solution you'd like

The solution that I found is when we update Mac OS to  Ventura then some Xcode terms changed and some setup files is missing for that we have to mannually install again that files and accept the terms.

So here the steps that I thinking to add into wiki as solution of this bug:

  ```
   xcode-select --install
   sudo xcodebuild -licence
   sudo xcode-select -r 
   sudo xcode-select -s /Library/Developer/CommandLineTools
   xcode-select -p 
  ```
After this Build bazel with this command:

```
bazel clean --expunge
bazel build //:oppia --noexperimental_check_desugar_deps
```

### Steps To Reproduce

1. Setup Bazel in Mac OS M1 - version Ventura 
2. Run Bazel build command

### Expected Behavior

Bazel build successful

### Screenshots/Videos

_No response_

### What device/emulator are you using?

_No response_

### Which Android version is your device/emulator running?

_No response_

### Which version of the Oppia Android app are you using?

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by MohitGupta121**
## Explanation

Fix #5094 : [Wiki update] Bazel build failed in mac m1

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Review by MohitGupta121 — COMMENTED** [review]
@adhiamboperes @BenHenning PTAL, Thanks!

File: wiki/Troubleshooting-Installation.md [hunk 1]
```diff
@@ -79,11 +84,33 @@ Here are some general troubleshooting tips for oppia-android. The specific platf
...
 
 
+4. If you encounter this error while building bazel in Mac M1:
+      ```
+      ERROR: /Users/OpenSource/oppia-android/model/src/main/proto/BUILD.bazel:167:20: Generating JavaLite proto_library //model/src/main/proto:profile_proto failed: (Segmentation fault): protoc failed: error executing command bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/protoc '--proto_path=bazel-out/android-armeabi-v7a-fastbuild/bin/model/src/main/proto/_virtual_imports/languages_proto' ... (remaining 8 argument(s) skipped)
+
+      Use --sandbox_debug to see verbose messages from the sandbox protoc failed: error executing command bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_google_protobuf/protoc '--proto_path=bazel-out/android-armeabi-v7a-fastbuild/bin/model/src/main/proto/_virtual_imports/languages_proto' ... (remaining 8 argument(s) skipped)
+      ```
+
```
Comment (adhiamboperes) [inline wiki/Troubleshooting-Installation.md:100]: I think this needs an explanation
```suggestion
Bazel requires Xcode commandline tools to build on M1, and the Xcode license also needs to be accepted.
```

File: wiki/Troubleshooting-Installation.md [hunk 2]
```diff
@@ -79,11 +84,33 @@ Here are some general troubleshooting tips for oppia-android. The specific platf
...
+   **Follow these steps to solve this error:**
+      ```
+      xcode-select --install
+      sudo xcodebuild -licence
+      sudo xcode-select -r 
+      sudo xcode-select -s /Library/Developer/CommandLineTools
+      xcode-select -p
```
Comment (adhiamboperes) [inline wiki/Troubleshooting-Installation.md]:
```suggestion
   **Follow these steps to solve this error:**
     - Install the commandline tools
      `xcode-select --install`
      - Accept the Xcode licence
      `sudo xcodebuild -licence`
      - Reset the xcode select path
      `sudo xcode-select -r `
      - Set the xcode select path to use CommandLineTools
      `sudo xcode-select -s /Library/Developer/CommandLineTools`
      - Confirm that the path was correctly set. The expected output is `/Library/Developer/CommandLineTools`
      `xcode-select -p` 
```

File: wiki/Troubleshooting-Installation.md [hunk 3]
```diff
@@ -79,11 +84,33 @@ Here are some general troubleshooting tips for oppia-android. The specific platf
...
+      sudo xcodebuild -licence
+      sudo xcode-select -r 
+      sudo xcode-select -s /Library/Developer/CommandLineTools
+      xcode-select -p 
+      ```
+   After you successfully run above commands, build app using Bazel by following command:
+      ```
+      bazel clean --expunge
+      bazel build //:oppia --noexperimental_check_desugar_deps
+      ```
```
Comment (adhiamboperes) [inline wiki/Troubleshooting-Installation.md:120]: Add: the `--noexperimental_check_desugar_deps` flag is explained in the [bazel blog](https://blog.bazel.build/2018/12/19/bazel-0.21.html#android).

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @MohitGupta121, I think some more explanation is needed, as I have suggested.

File: wiki/Troubleshooting-Installation.md (hunk 1 above)
Comment (MohitGupta121) [reply wiki/Troubleshooting-Installation.md:100]: Done.

File: wiki/Troubleshooting-Installation.md (hunk 2 above)
Comment (MohitGupta121) [reply wiki/Troubleshooting-Installation.md]: Done.

File: wiki/Troubleshooting-Installation.md (hunk 3 above)
Comment (MohitGupta121) [reply wiki/Troubleshooting-Installation.md:120]: Done.

**Review by MohitGupta121 — COMMENTED** [review]
@adhiamboperes Thanks you so much, PTAL.

File: wiki/Troubleshooting-Installation.md [hunk 4]
```diff
@@ -79,11 +84,42 @@ Here are some general troubleshooting tips for oppia-android. The specific platf
...
+
+- Reset the xcode select path: `sudo xcode-select -r `
+
+ - Set the xcode select path to use CommandLineTools: `sudo xcode-select -s /Library/Developer/CommandLineTools`
+
+- Confirm that the path was correctly set. The expected output is: `/Library/Developer/CommandLineTools`
+
+      xcode-select -p
+
+After you successfully run above commands, build app using Bazel by following command:
```
Comment (adhiamboperes) [inline wiki/Troubleshooting-Installation.md]:
```suggestion
After successfully running the above commands, build the app using Bazel by running the following command:
```

**Review by adhiamboperes — COMMENTED** [review]
Thanks @MohitGupta121, just one grammar suggestion.

File: wiki/Troubleshooting-Installation.md (hunk 4 above)
Comment (MohitGupta121) [reply wiki/Troubleshooting-Installation.md]: Done

**Review by MohitGupta121 — COMMENTED** [review]
@adhiamboperes apply changes, PTAL, Thanks!

**Review by adhiamboperes — APPROVED** [review]
Thanks, LGTM

**Pull request merged**